### PR TITLE
feat(providers): add tls options to nodemailer

### PR DESCRIPTION
**Pull request by p-fernandez**
### What change does this PR introduce? 

 
Introduces the ability for the users to override and set up TLS configurations.

### Why was this change needed?

To give configuration power to SMTP providers for our users and be able to integrate Novu.

### CODE REVIEW

**Comment by jainpawan21** [comment]
Can we merge this PR?

**Comment by p-fernandez** [comment]
> Can we merge this PR?

Work still in draft. Pending of completing tests and test it manually somehow.

**Review by scopsy — APPROVED** [review]

File: apps/web/cypress/tests/integration-store.spec.ts [hunk 1]
```diff
@@ -5,46 +5,84 @@ describe('Integration store page', function () {
...
+    it('should display use credentials on settings modal', function () {
+      interceptNodemailerIntegration(true);
 
-    cy.getByTestId('apiKey').should('have.value', '123');
-    cy.getByTestId('from').should('have.value', 'cypress');
+      cy.visit('/integrations');
+
+      getFirstIntegrationCard().getByTestId('provider-card-settings-svg').click();
+
+      cy.getByTestId('apiKey').should('have.value', '123');
```
Comment (ainouzgali) [inline apps/web/cypress/tests/integration-store.spec.ts:75]: @p-fernandez seems that nodemailer doesn't have apikey in provider-credentials

Comment (p-fernandez) [reply apps/web/cypress/tests/integration-store.spec.ts:75]: Thanks for spotting it. Fixed in a different PR.

**Pull request merged**